### PR TITLE
refactor: ziofy ResourcesRouteV2 (DEV-2091)

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number

Issue Number:  (DEV-2091)

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] fix: represents bug fixes
- [x] refactor: represents production code refactoring
- [ ] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [ ] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] Maybe (not 100% sure => check with FE)

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

**Review by irinaschubert — APPROVED** [review]
looks good, just a small comment on a docstring

**Review by mpro7 — APPROVED** [review]

**Review by BalduinLandolt — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV2.scala [hunk 1]
```diff
@@ -210,45 +209,31 @@ object RouteUtilV2 {
...
+   * Gets the required project IRI specified in a Knora-specific HTTP header.
    *
-   * @param ctx The akka-http [[RequestContext]].
-   *
-   * @param stringFormatter An instance of the [[StringFormatter]].
-   * @return The [[SmartIri]] contains the specified project IRI.
+   * @param requestContext The akka-http [[RequestContext]].
+   * @return The specified project IRI, or [[None]] if no project header was included in the request.
+   *         Fails with a [[BadRequestException]] if the project IRI is invalid.
+   *         Fails with a [[BadRequestException]] if the project header is missing.
```
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV2.scala]: You are right. Fixed in e6fe0c9493f0ad3cd151ba9d67347cacb925d8cc

File: webapi/src/it/scala/org/knora/webapi/e2e/v1/ResourcesV1R2RSpec.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ class ResourcesV1R2RSpec extends R2RSpec {
   private val resourcesPathV1 =
     DSPApiDirectives.handleErrors(system, appConfig)(ResourcesRouteV1().makeRoute)
   private val resourcesPathV2 =
-    DSPApiDirectives.handleErrors(system, appConfig)(ResourcesRouteV2(routeData, runtime).makeRoute)
+    DSPApiDirectives.handleErrors(system, appConfig)(ResourcesRouteV2(appConfig).makeRoute)
```
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/it/scala/org/knora/webapi/e2e/v1/ResourcesV1R2RSpec.scala:60]: The `KnoraRouteData` is instantiated with the  identical `AppConfig` instance. I am aiming for removing the `KnoraRouteData` so it should rather be that in the other places the `AppConfig` instance is used directly.

[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/it/scala/org/knora/webapi/e2e/v1/ResourcesV1R2RSpec.scala:60]: Fixed in 941e76a7519e6c8b9f8f9dc289223d44f023a31f

**Pull request merged**